File: Advanced DSA/02. Array Techniques/04. Array Rotation/Solution.cpp
void reverse(vector<int> &A, int left, int right) {
    for (; left < right; ++left, --right)
        swap(A[left], A[right]);
}


vector<int> Solution::solve(vector<int> &A, int B) {
    int n = A.size();
    B %= n;
    reverse(A, 0, n - B - 1);
    reverse(A, n - B, n - 1);
    reverse(A, 0, n - 1);
    return A;
}

File: Advanced DSA/02. Array Techniques/04. Array Rotation/Solution.py
def reverse(A, left, right):
    while left < right:
        A[left], A[right] = A[right], A[left]
        left += 1
        right -= 1


class Solution:
    # @param A : list of integers
    # @param B : integer
    # @return a list of integers
    def solve(self, A, B):
        n = len(A)
        B %= n
        reverse(A, 0, n - B - 1)
        reverse(A, n - B, n - 1)
        reverse(A, 0, n - 1)
        return A
